onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -2366,7 +2376,7 @@ std::unique_ptr<IGraphPartitioner> IGraphPartitioner::CreateGraphPartitioner(con
                                                                              const PathString& config_file) {
   // use device based partitioner by default
   IGraphPartitioner::GraphPartitioningStrategy partitioner_type =
-      IGraphPartitioner::GraphPartitioningStrategy::Unknown;
+      IGraphPartitioner::GraphPartitioningStrategy::DeviceBasedPartition;
```

Comment:
why update this?

Reply:
To make the code neat and short

Reply:
This is fine, while the implication is bit different - with unknown type as initial value, we would later log "Use ... as default" to call out the fact that "hey man, you didn't tell so we take default as ...".